src/NServiceBus.Core/Transports/OutgoingMessage.cs

```diff
@@ -0,0 +1,32 @@
+namespace NServiceBus.Transports
+{
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// The message going out to the transport
+    /// </summary>
+    public class OutgoingMessage
+    {
+        /// <summary>
+        /// Constructs the message
+        /// </summary>
+        /// <param name="headers">The headers associated with this message</param>
+        /// <param name="body">The body of the message</param>
+        public OutgoingMessage(Dictionary<string, string> headers,byte[] body)
```

Comment:
>but what happens if the newed up message contains a stream

How does the current serializers handle this?

On Tue, Mar 10, 2015 at 9:50 AM, John Simons <notifications@github.com>
wrote:

> In src/NServiceBus.Core/Transports/OutgoingMessage.cs
> <https://github.com/Particular/NServiceBus/pull/2678#discussion_r26105624>
> :
>
> > @@ -0,0 +1,32 @@
> > +namespace NServiceBus.Transports
> > +{
> > +    using System.Collections.Generic;
> > +
> > +    /// <summary>
> > +    /// The message going out to the transport
> > +    /// </summary>
> > +    public class OutgoingMessage
> > +    {
> > +        /// <summary>
> > +        /// Constructs the message
> > +        /// </summary>
> > +        /// <param name="headers">The headers associated with this message</param>
> > +        /// <param name="body">The body of the message</param>
> > +        public OutgoingMessage(Dictionary<string, string> headers,byte[] body)
>
> Good point Andreas, but what happens if the newed up message contains a
> stream, and we are not using the databus because the max data is only
> 500kb? I know it is a pretty stretched case but it can happen. I do believe
> with the new abstraction we should aim to try to use streams because it
> will give us the flexibility later on to do much more.
> … <#14c02e0fd10f2f79_>
> On Tuesday, March 10, 2015, Andreas Öhlund <notifications@github.com>
> wrote: In src/NServiceBus.Core/Transports/OutgoingMessage.cs <
> https://github.com/Particular/NServiceBus/pull/2678#discussion_r26103351>
> : > @@ -0,0 +1,32 @@ > +namespace NServiceBus.Transports > +{ > + using
> System.Collections.Generic; > + > + /// <summary> > + /// The message going
> out to the transport > + /// </summary> > + public class OutgoingMessage >
> + { > + /// <summary> > + /// Constructs the message > + /// </summary> > +
> /// <param name="headers">The headers associated with this message</param>
> > + /// <param name="body">The body of the message</param> > + public
> OutgoingMessage(Dictionary<string, string> headers,byte[] body) Is it? With
> the databus data is stored on "disk" and streamed into memory, all good.
> This is outgoing messages so it will already be in memory since the users
> will have "new'ed" up the message? (so you will already have allocated all
> the memory) … <#14c02a8c6d9aa85e_> On Tue, Mar 10, 2015 at 8:44 AM, John
> Simons <notifications@github.com <javascript:_e(%7B%7D,'cvml','
> notifications@github.com');>> wrote: In
> src/NServiceBus.Core/Transports/OutgoingMessage.cs <
> https://github.com/Particular/NServiceBus/pull/2678#discussion_r26103254>
> : > @@ -0,0 +1,32 @@ > +namespace NServiceBus.Transports > +{ > + using
> System.Collections.Generic; > + > + /// <summary> > + /// The message going
> out to the transport > + /// </summary> > + public class OutgoingMessage >
> + { > + /// <summary> > + /// Constructs the message > + /// </summary> > +
> /// <param name="headers">The headers associated with this message</param>
> > + /// <param name="body">The body of the message</param> > + public
> OutgoingMessage(Dictionary<string, string> headers,byte[] body) I'm
> confused, what are u unclear about? This is exactly the same issue as the
> issue with the databus! … <#14c02a4c511f2712_> On Tuesday, March 10, 2015,
> Andreas Öhlund <notifications@github.com <javascript:_e(%7B%7D,'cvml','
> notifications@github.com');>> wrote: In
> src/NServiceBus.Core/Transports/OutgoingMessage.cs <
> https://github.com/Particular/NServiceBus/pull/2678#discussion_r26103000>
> : > @@ -0,0 +1,32 @@ > +namespace NServiceBus.Transports > +{ > + using
> System.Collections.Generic; > + > + /// <summary> > + /// The message going
> out to the transport > + /// </summary> > + public class OutgoingMessage >
> + { > + /// <summary> > + /// Constructs the message > + /// </summary> > +
> /// <param name="headers">The headers associated with this message</param>
> > + /// <param name="body">The body of the message</param> > + public
> OutgoingMessage(Dictionary<string, string> headers,byte[] body)
> @johnsimons <https://github.com/johnsimons> <https://github.com/johnsimons>
> <https://github.com/johnsimons> <https://github.com/johnsimons> I could
> go both ways. How about we make this out of scope for this pull and you and
> me spike this when you're back next week? — Reply to this email directly or
> view it on GitHub <
> https://github.com/Particular/NServiceBus/pull/2678/files#r26103000>. —
> Reply to this email directly or view it on GitHub <
> https://github.com/Particular/NServiceBus/pull/2678/files#r26103254>. —
> Reply to this email directly or view it on GitHub <
> https://github.com/Particular/NServiceBus/pull/2678/files#r26103351>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Particular/NServiceBus/pull/2678/files#r26105624>.
>